opensearch_py_ml/ml_commons/ml_commons_client.py

```diff
@@ -359,6 +360,36 @@ def _get_task_info(self, task_id: str):
             url=API_URL,
         )
 
+    def search_task(self, **kwargs) -> object:
```

Comment:
Sorry for making this complicated for you. Let's make this completely simple. 

1. We will expect a json string, similar like [this](https://opensearch.org/docs/2.4/ml-commons-plugin/api/#search-task) which will be a query body.

2. we will check if the query body is a valid json or not. We can check this :

import json

```
def is_json(myjson):
  try:
    json.loads(myjson)
  except ValueError as e:
    return False
  return True
```
3. And if the json is valid then we will simple call the api: `GET /_plugins/_ml/tasks/_search`.

Let's keep this simple.

I wanted to make this completely dynamic so that customer doesn't need to build the query by himself, but it seems like it can be bit complex and we might not cover all the edge cases. So better to expect a valid search query body to make things simple.

Reply:
Hello. Sorry for dragging out this task for so long. I was worked in my bachelor diploma work. Now, I am free and I am already working on it. As promised I will do this task.

Reply:
No worries. Thanks for working on it. Please let me know if you have any question or concern.

Reply:
I will try to finish before Sunday and make it dynamic so customer does not need to build the query by himself. Anyway if it will not work I will just make just json input for customer from scratch.

Reply:
Sounds good. May be you can take a look at [this article](https://www.elastic.co/guide/en/elasticsearch/reference/current/search-your-data.html#common-search-options) to understand more options about the search in elastic search or opensearch.

Reply:
I collected all url parameters and main request body parameters of search api. I made helper functions for building term query, range, match, bool query, geo, term level queries. Can I finish collecting all search api request body parameters till thursday with testing?

Reply:
Example of helper function that I made: 
def range_query(field, gte=None, gt=None, lte=None, lt=None):
    query = {"range": {field: {}}}
    if gte is not None:
        query["range"][field]["gte"] = gte
    if gt is not None:
        query["range"][field]["gt"] = gt
    if lte is not None:
        query["range"][field]["lte"] = lte
    if lt is not None:
        query["range"][field]["lt"] = lt
    return query

Reply:
I feel like to cover all the use cases in a single method, we might need to have lot of parameters and this might get confusing for users also. May be we should look for the simpler solution as this query pattern is pretty well used by users? What do you think?

Reply:
Ok

Reply:
Function and testing is done. From syncing with latest changes I have several "merge branch" commits. Is it ok?

Reply:
I will rise PR again. I broke rebasing.